lib/effects/scale_effect.dart

```diff
@@ -4,51 +4,39 @@ import 'package:meta/meta.dart';
 import '../extensions/vector2.dart';
 import './effects.dart';
 
-double _direction(double p, double d) => (p - d).sign;
-double _length(double a, double b) => (a - b).abs();
-
 class ScaleEffect extends PositionComponentEffect {
   Vector2 size;
   double speed;
   Curve curve;
-
-  Vector2 _original;
-  Vector2 _diff;
-  final Vector2 _dir = Vector2.zero();
+  Vector2 _startSize;
+  Vector2 _delta;
 
   ScaleEffect({
     @required this.size,
     @required this.speed,
     this.curve,
     isInfinite = false,
     isAlternating = false,
+    isRelative = false,
     Function onComplete,
-  }) : super(isInfinite, isAlternating, onComplete: onComplete);
+  }) : super(isInfinite, isAlternating,
```

Comment:
one per line with trailing commas or all in one line :p

Reply:
Good catch, fixed!